docs/sources/flow/reference/components/otelcol.processor.k8sattributes.md

```diff
@@ -0,0 +1,274 @@
+exclude | [exclude][] | Exclude pods from being processed | no
+exclude > pod | [pod][] | Pod information | no
+
+
+The `>` symbol indicates deeper levels of nesting. For example, `extract > annotation`
+refers to an `annotation` block defined inside an `extract` block.
+
+[output]: #output-block
+[extract]: #extract-block
+[extract_field]: #extract-field-block
+[filter]: #filter-block
+[filter_field]: #filter-field-block
+[pod_association]: #pod-association-block
+[pod_association_source]: #pod-association-source-block
+[exclude]: #exclude-block
+[pod]: #pod-block
+
+### Extract block
+
+The `extract` block configures which metadata, annotations and labels to extract from the pod, and to add to the spans.
+
+The following attributes are supported:
+
+Name | Type           | Description                                                                                                                                                                                                                                                 | Default | Required
+---- |----------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------| ------- | --------
+`metadata` | `list(string)` | Pre-configured metadata keys to add. See [k8sattributeprocessor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/processor/k8sattributesprocessor/v0.85.0/processor/k8sattributesprocessor/config.go#L119) for the list of keywords. |  | no
+
+
+### Extract field block
+
+The `label` or `annotation` block configures which metadata or labels to extract from the pod and add to the spans.
+
+The following attributes are supported:
+
+Name | Type           | Description                                                                                           | Default | Required
+---- |----------------|-------------------------------------------------------------------------------------------------------|---------| --------
+`tag_name` | `list(string)` | TagName represents the name of the resource attribute that will be added to logs, metrics or spans.   |         | no
+`key` | `list(string)` | Key represents the annotation (or label) name. This must exactly match an annotation (or label) name. |         | no
+`key_regex` | `list(string)` | KeyRegex is a regular expression used to extract a Key that matches the regex.                   |         | no
+`regex` | `list(string)` | Regex is an optional field used to extract a sub-string from a complex field value.                                                                                                  |         | no
+`from` | `list(string)` | From represents the source of the labels/annotations. Allowed values are "pod" and "namespace".                                                | `pod`    | no
```

Comment:
```suggestion
`tag_name` | `list(string)` | The name of the resource attribute that will be added to logs, metrics or spans.  | `[]` | no
`key` | `list(string)` | The annotation (or label) name. This must exactly match an annotation (or label) name. | `[]` | no
`key_regex` | `list(string)` | A list of regular expressions used to extract a "key" that matches the regex.    | `[]` | no
`regex` | `list(string)` | Configures how to extract a sub-string from a complex field value.                                                                                                  | `[]` | no
`from` | `list(string)` | Represents the source of the labels/annotations.  | `pod` | no

When `tag_name` is not specified, a default tag name will be used of the format:
* `k8s.pod.annotations.<annotation key>`
* `k8s.pod.labels.<label key>`

For example, if `tag_name` is not specified and the key is `git_sha`,
then the attribute name will be `k8s.pod.annotations.git_sha`.

Either the `key` attribute or the `key_regex` attribute should be set, not both.

When `key_regex` is present, `tag_name` supports back reference to both
named capturing and positioned capturing.
For example, if your pod spec contains the following labels...

* `app.kubernetes.io/component: mysql`
* `app.kubernetes.io/version: 5.7.21`

...and you'd like to add tags for all labels with prefix `app.kubernetes.io/` and also trim the prefix,
then you can specify the following extraction rules:

```river
extract {
	label {
		key_regex = "kubernetes.io/(.*)"
		tag_name  = "$$1"
	}
}
\`\`\`

this will add the `component` and `version` tags to the spans or metrics.

The `from` attribute can be set to either `"pod"` or `"namespace"`.
```

I got this example from [config.go](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.80.0/processor/k8sattributesprocessor/config.go#L114), but I'm actually not sure if `"$$1"` should be `"$1"` instead? Also, it might be better to include such an example in the "examples" section, and just link to it from here.

Also, the "\`\`\`" should not include the \, but I wasn't how to include this in the GitHub comment.

It may be good to include an example for the "regex" attribute too, like the one [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.80.0/processor/k8sattributesprocessor/config.go#L128-L147).

Reply:
Thank you! Yeah it should be $1 with River. I'm working through all the changes requested now. Thank you for the thorough feedback.